tools/android_lint/bin/main.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:path/path.dart' as path;
+import 'package:process/process.dart';
+
+/// Runs the Android SDK Lint tool on flutter/shell/platform/android.
+///
+/// This script scans the flutter/shell/platform/android directory for Java
+/// files to build a `project.xml` file.  This file is then passed to the lint
+/// tool and HTML output is reqeusted in the directory for the `--out`
+/// parameter, which defaults to `lint_report`.
+///
+/// The `--in` parameter may be specified to to force this script to scan a
+/// specific location for the engine repository, and expects to be given the
+/// `src` directory that contains both `third_party` and `flutter`.
+///
+/// At the time of this writing, the Android Lint tool doesn't work well with
+/// Java > 1.8.  This script will print a warning if you are not running
+/// Java 1.8.
+Future<void> main(List<String> args) async {
```

Comment:
Can we break this function down?  Something like:

```dart
request = parseArgs(args);
writeProjectFile(request);
result = runLint();
return result;
```

Reply:
More or less done - I could definitely get fancier with this, but right now I just want something that works.